code/modules/power/gravitygenerator.dm

```diff
@@ -193,9 +193,14 @@ GLOBAL_LIST_EMPTY(gravity_generators)
 		part.sprite_number = count
 		part.main_part = src
 		generator_parts += part
+		RegisterSignal(part, COMSIG_QDELETING, PROC_REF(on_part_qdeleted))
```

Comment:
This should be handled in `/obj/machinery/gravity_generator/part/Destroy` instead
(`main_part.parts -= src`)

They already hold a ref for us, they can handle cleaning up for us as well